js/app/dashboard/redux/actions.js

```diff
@@ -680,7 +683,135 @@ export function startTask(task) {
   }
 }
 
-export function completeTask(task, notes = '', success = true) {
+/**
+ * Creates a task queue processor with the given context
+ * @returns {Function} - Configured task queue processor
+ */
+function createTaskProcessor(context) {
```

Comment:
To be honest, this looks a bit like "spaghetti code," especially a callback `onRequireConfirmation` that returns a promise that passes `onResolve` inside a redux action... 
Are you doing all this because `TaskModalContent` is a class component?

Maybe we can: 

- move a call to `completeTask` inside `TaskCompleteForm`
- re-implement `completeTask` using an RTK query mutation hook: https://github.com/coopcycle/coopcycle-web/blob/51c13a0c04eb90d19b5aa9be3c243bdcbaa73673/js/app/api/slice.js#L4 https://redux-toolkit.js.org/rtk-query/usage/mutations#mutation-hook-behavior
- show `TaskConfirmationDialog` conditionally based on `completeTask` response
- mark pending pickups as done directly from `TaskConfirmationDialog` (by dispatching an action that will trigger the same queue-like behaviour)